peps/pep-0740.rst

```diff
@@ -268,47 +441,62 @@ following structure:
-  ``attestations`` field, as described in :ref:`upload-endpoint`.
+  illustrative example of a ``publisher`` object is provided in :ref:`appendix-1`.
+* ``attestations`` is a **required** JSON array, containing one or
+  more attestation objects. This array is a superset of ``attestations`` array
+  supplied by the uploader through the ``attestations`` field at upload time,
+  as described in :ref:`upload-endpoint`.
 
   Because ``attestations`` is a superset of the file's original uploaded attestations,
-  the index **MAY** chose to embed additional attestations of its own.
+  the index **MAY** chose to embed additional attestations. These attestations
+  may come from the index itself, from third parties, or as subsequent actions
+  produced by the original uploader.
 
-Simple Index
-^^^^^^^^^^^^
+.. _attestation-verification:
 
-* When an uploaded file has one or more attestations, the index **MAY** include a
-  ``data-provenance`` attribute on its file link, with a value of either
-  ``true`` or ``false``.
-* When ``data-provenance`` is ``true``, the index **MUST** serve a
-  :ref:`provenance object <provenance-object>` at the same URL, but with
-  ``.provenance`` appended to it. For example, if ``HolyGrail-1.0.tar.gz``
-  exists and has associated attestations, those attestations would be located
-  within the provenance object hosted at ``HolyGrail-1.0.tar.gz.provenance``.
+Attestation verification
+------------------------
 
-JSON-based Simple API
-^^^^^^^^^^^^^^^^^^^^^
-
-* When an uploaded file has one or more attestations, the index **MAY** include a
-  ``provenance`` object in the ``file`` dictionary for that file.
-* ``provenance``, when present, **MUST** be a :ref:`provenance object <provenance-object>`.
+Verifying an attestation object requires verification of each of the following:
 
-These changes require a version change to the JSON API:
+* ``verification_material.certificate`` is a valid signing certificate, as
+  issued by an *a priori* trusted authority (such as a root of trust already
+  present within the verifying client).
+* ``verification_material.certificate`` identifies an appropriate signing
+  subject, such as the machine identity of the Trusted Publisher that published
+  the package.
+* ``message_signature`` can be verified by ``verification_material.certificate``,
+  using the reconstructed attestation payload as the cleartext input. The
+  verifier **MUST** reconstruct the attestation payload itself.
 
-* The ``api-version`` must specify version 1.2 or later.
+In addition to the above required steps, a verifier **MAY** additionally verify
+``verification_material.transparency_entries`` on a policy basis, e.g. requiring
+at least one transparency log entry or a threshold of entries. When verifying
+transparency entries, the verifier **MUST** confirm that the inclusion time for
+each entry lies within the signing certificate's validity period.
 
```

Comment:
Do we need a description of a trusted root file?

Reply:
I *think* we don't -- the reasoning I'm applying here is that the trust root is essentially an implementation detail in the layer of "you need an a priori source of trust," similar to how CABF doesn't specify anything about OS roots other than that they have the basic trust anchor contents needed to construct chains.